simpler_setup/runtime_compiler.py

```
@@ -303,20 +339,23 @@ def _build(actual_build_dir: str) -> Union[bytes, Path]:
                     dest_dir = Path(dispatcher_dest)
                     dest_dir.mkdir(parents=True, exist_ok=True)
                     dest_dispatcher = dest_dir / dispatcher_name
-                    shutil.copy2(dispatcher_so, dest_dispatcher)
                     # Cross-arch strip: aicpu .so is aarch64 even on x86 host;
                     # GNU strip 2.38 on Ubuntu 22.04 cannot read it. Prefer
                     # llvm-strip (multi-arch) when available.
                     strip_bin = shutil.which("llvm-strip") or shutil.which("aarch64-linux-gnu-strip") or "strip"
-                    subprocess.run([strip_bin, "-s", str(dest_dispatcher)], check=True)
+
+                    def _strip(staged: Path, strip_bin: str = strip_bin) -> None:
+                        subprocess.run([strip_bin, "-s", str(staged)], check=True)
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)